src/app/code/community/Ambimax/PriceImport/Helper/Data.php

```diff
@@ -117,11 +118,172 @@ public function getErpImporter(bool $load = false): Ambimax_PriceImport_Model_Er
+
+    }
+
+    protected function getFtpServerAddress(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_host');
+
+    }
+
+    protected function getFtpFilesLocation(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_path');
+
+    }
+
+    protected function getFtpFileNames(): string
+    {
+
+        return Mage::getStoreConfig('ambimax_priceimport/erp_import_options/file_sftp_name');
+
+    }
+
+    /**
+     * @param $originalFilenames
+     */
+    public function prepareFileNames($originalFilenames)
+    {
+        if (empty($originalFilenames)) {
+            return;
+        }
+        if (!empty($originalFilenames)) {
+            foreach ($originalFilenames as $name) {
+                $filenames[$name] = substr($name, strripos($name, '/') + 1);
+            }
+        }
+
+        return $filenames;
+    }
+
+    public function getNewestFiles(array $allFileNames): array
+    {
+        $fileName = $this->cleanFileNames($allFileNames);
+        $onlyPetfriendsFileName = $this->cleanFileNames($allFileNames, 1);
+        $fileNames = [];
+        array_push($fileNames, $fileName, $onlyPetfriendsFileName);
+        return $fileNames;
+    }
+
+    /*
+     * use $hasSuffix = 1 for list with suffix
+     */
+    public function cleanFileNames(array $fileNames, int $hasSuffix = 0): string
+    {
+        if ($hasSuffix) {
+            $fileNames = array_filter($fileNames, array($this, 'isOnlyPetfriendsFile'));
+        } else $fileNames = array_filter($fileNames, array($this, 'isPriceFile'));
+
+        sort($fileNames);
+        $newestFileName = array_pop($fileNames);
+        if (!empty($fileNames)) {
+            $this->cleanFtpFiles($fileNames);
+        }
+        return $newestFileName;
+    }
+
+    public function cleanFtpFiles(array $fileNames): void
+    {
+        foreach ($fileNames as $filename) {
+            $this->deleteFtpFiles($this->getFtpFilesLocation() . $filename);
+        }
+    }
+
+    public function deleteFtpFiles($filePath): void
+    {
+        $connectionInforamtions = $this->_getConnectionInformations();
+
+        //@codingStandardsIgnoreStart
+        $connectionId = ftp_connect($connectionInforamtions['serverAddress']);
+        if (!ftp_login($connectionId, $connectionInforamtions['username'], $connectionInforamtions['password'])) {
+            $this->throwLoginError($connectionInforamtions['serverAddress'], $connectionInforamtions['username']);
+        }
+        ftp_delete($connectionId, $filePath);
+        ftp_close($connectionId);
+        //@codingStandardsIgnoreEnd
+    }
+
+    public function isPriceFile(string $fileName): bool
+    {
+        $fileNameSetting = $this->getFtpFileNames();
+        if (!str_starts_with($fileName, $fileNameSetting)) {
+            return false;
+        }
+        if (!str_starts_with($fileName, $fileNameSetting . $this->_listsuffix)) {
```

Comment:
Is it really something like `price-list-file-name.csv-fwpf-`? Seems weird to me.

Reply:
its artikelstamm-fwpf-20210622010002

Reply:
you only use the filename start in the administration fields

Reply:
Aaaahhh